File: HeartBeatSystemDemo/HeartBeatSystemDemo/src/HBSThread.java
import java.util.Timer;
import java.util.TimerTask;

/**
 * 使用继承Runnable接口的方法创建多线程
 * 将这个类实例化让后放到Thread实例的构造器中，再让Thread实例调用start()方法
 * 多线程创建成功！
 */
public class HBSThread implements Runnable{
    Timer timer = new Timer();
    @Override
    public void run() {
        timer.schedule(new TimerTask(),0 , 2000);
    }
    private class TimerTask extends java.util.TimerTask{
        private HeartBeatSystem hbs = new HeartBeatSystem();

        @Override
        public void run() {
            hbs.randomChangeHeartBeat();
            hbs.updateHeartBeat();
            System.out.println(hbs.getHeartBeatRate());
        }
    }
}

File: UnityProjet/scripts/HeartBeatSystem/thread/HBSThread.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Threading;
using UnityEngine;

public class HeartBeatThread
{
    private HeartBeatSystem HBS;
    private Boolean IsPause;
    public HeartBeatThread(HeartBeatSystem hbs)
    {
        this.HBS = hbs;
        IsPause = false;
    }
    public void Pause()
    {
        IsPause = true;
    }
    public void UnPause()
    {
        IsPause = false;
    }
    public void Run() 
    {
        while (true)
        {
            if (IsPause)
            {
                continue;
            }
            Update();
            Thread.Sleep(2000);
        }
    }
    public void Update()
    {
        HBS.RandomrandomChangeHeartBeat();
        HBS.UpdateHeartBeat();
        Debug.Log(HBS.GetHeartBeatRate());
    }
}
